### ISSUE TITLE
Remaining minor issues

### ISSUE THREAD

**Issue by ryuudraco** [ryuudraco/MATCHAR#7]
**_Profile: (not high priority)_**
Image upload needs to reject anything that is not an image (a pdf for example needs to give an error message)

**_History:_**
Create backend for history so that a user can see who has liked/visited him, example
This (person) has viewed your profile.
This (person) has liked your profile.

I don't know if on backend or not, but do a quick check to see if two users like each other. If they do they Match and in their history will be a message saying You have matched with (person)!

**_Fame rating:_** 
Decide how to do a public fame rating, perhaps number of likes divided by number of views of profile? idk

**_Profile Suggestion:_**
Suggest profiles based on location/fame rating/common tags. This is pretty much the browse except we just need to sort them by location/fame rating/common tags.

**_Advanced Search:_**
User must be able to run an advanced search based on age gap, fame rating group, location, one or multiple interest tags

Users must be sorted by Age/location/fame/tags.

**_Report/Blocking:_**
User must be able to report/block another user preventing them from appearing in their list of suggested users

_**Messages:**_
Messages need to be working. Can have a 10 second delay.

_**Notifications:**_
A user must receive a notification when they get a message, a like, or a match. Can have a 10 second delay.

**Comment by ryuudraco** [comment]
@WeyIin

**Comment by ryuudraco** [comment]
Front-end has been completed for the following until I get more values/variables for loops/if statements.
I have a few ideas regarding some of these to get them done quickly. Especially **Report/Block** and **Fame Rating**

History:
Create backend for history so that a user can see who has liked/visited him, example
This (person) has viewed your profile.
This (person) has liked your profile.

I don't know if on backend or not, but do a quick check to see if two users like each other. If they do they Match and in their history will be a message saying You have matched with (person)!

Fame rating:
Decide how to do a public fame rating, perhaps number of likes divided by number of views of profile? idk

Profile Suggestion:
Suggest profiles based on location/fame rating/common tags. This is pretty much the browse except we just need to sort them by location/fame rating/common tags.

Report/Blocking:
User must be able to report/block another user preventing them from appearing in their list of suggested users

Messages:
Messages need to be working. Can have a 10 second delay.

Notifications:
A user must receive a notification when they get a message, a like, or a match. Can have a 10 second delay.

**Comment by ryuudraco** [comment]
I have created the BlockDAO and beans files as we ll as a blocks database. The Idea for the blocks is just do a copy pasta of the likes (with the counting total for a reason which I will explain in a bit), so that we have most of it done. The only part of it that will not be done is the "bad mark" on said persons profile that will hide it from whoever has blocked them. If they get blocked by 5-10 people for being "fake", "scammy" or just a douchebag in general then a report email is sent out saying your profile has been reported and your profile will receive a "red mark" something on the frontend that displays to other users you have been blocked by 5 people.

**Comment by ryuudraco** [comment]
I have created a fame_rating table if it is necessary I don't think it will be however as it will be something super simple. I didn't know if you wanted to create a DAO/Bean for it so I left that out however this will be a really simple task. The fame rating will pretty much be at default 1, the way we will calculate this from then on is by how many likes/blocks you have. If your profile gets liked you will receive 1 point (your fame rating is now 2) if you get blocked you will lose 1 point (your fame rating is now back at 1)

**Comment by ryuudraco** [comment]
I have created buttons on the browse section (which is the suggest profiles) to sort by Age, Location, Tags. What we can do here is somewhat simple but pretty much when they click age it will just grab everyone and place them in incremental order from youngest to oldest. Location is pretty much going to do 3 checks if their country matches yours they get a +1, if their province matches yours they get a +1 if their city matches they get a +1 if any of these don't match they don't get a point, i.e. if my country and province match then I get a rating of 2. We can take these ratings and order them by those ratings. i.e. if all 3 match my card is at the top and sorted alphabetically so all those with 3 points will be up there. If they only have 2 points then they come after the people with 3 points and after those with 2 points will be the profiles with 1 then 0.

**Comment by ryuudraco** [comment]
History, I'm not sure what we will need to do here. I know we will be grabbing their likes if they liked you and just add it to the history however I'm not sure about the others, we will need to do a check if two people like each other and if they do they match and the message is there. Idk how to do the view one though I apologize.

**Comment by ryuudraco** [comment]
Messages and notifications are pretty much the same thing as before. We will just do json requests, every how ever many seconds we will do json requests for the messages and history. If a person has liked your profile and that json requests returns true. You will get that notification until you refresh your page or change pages. If a person has matched with you in history then the same thing happens a json request with a notification until refresh or change page. If a person has messaged you then kerblammo notification from json request lol. Very simple notification message that won't include any user data just message that they have a message/match/like.

**Comment by ryuudraco** [comment]
We can scrap advanced search, the low priority image validation is not really an issue. IT was more a comment the evaluator made so doesn't really need to be done as far as I'm concerned as there was not really a mention of security. If you want to do it you can though. Super low priority.
